datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java

```diff
@@ -77,33 +86,38 @@ public SdxClusterResponse create(@ValidStackNameFormat @ValidStackNameLength Str
     }
 
     @Override
-    public FlowIdentifier delete(String name, Boolean forced) {
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.WRITE)
+    public FlowIdentifier delete(@ResourceName String name, Boolean forced) {
         String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
         return sdxService.deleteSdx(userCrn, name, forced);
     }
 
     @Override
-    public FlowIdentifier deleteByCrn(String clusterCrn, Boolean forced) {
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.WRITE)
+    public FlowIdentifier deleteByCrn(@ResourceCrn String clusterCrn, Boolean forced) {
         String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
         return sdxService.deleteSdxByClusterCrn(userCrn, clusterCrn, forced);
     }
 
     @Override
-    public SdxClusterResponse get(String name) {
+    @CheckPermissionByAccount
+    public SdxClusterResponse get(@ResourceName String name) {
         String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
         SdxCluster sdxCluster = sdxService.getSdxByNameInAccount(userCrn, name);
         return sdxClusterConverter.sdxClusterToResponse(sdxCluster);
     }
 
     @Override
-    public SdxClusterResponse getByCrn(String clusterCrn) {
+    @CheckPermissionByAccount
+    public SdxClusterResponse getByCrn(@ResourceCrn String clusterCrn) {
         String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
         SdxCluster sdxCluster = sdxService.getByCrn(userCrn, clusterCrn);
         return sdxClusterConverter.sdxClusterToResponse(sdxCluster);
     }
 
     @Override
-    public List<SdxClusterResponse> getByEnvCrn(@ValidCrn String envCrn) {
+    @CheckPermissionByAccount
+    public List<SdxClusterResponse> getByEnvCrn(@ResourceCrn @ValidCrn String envCrn) {
```

Comment:
This is an environment CRN not a Datalake CRN.
Would this API call even work?

Reply:
hmm, that is a good question, i have to think about this scneario, anyway i'll remove resourcecrn from here for now

Reply:
added to TODO list